src/Action/Order/MercadopagoPreference.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace AppBundle\Action\Order;
+
+use MercadoPago;
+use AppBundle\Service\SettingsManager;
+use Symfony\Component\HttpFoundation\Request;
+use AppBundle\Payment\MercadopagoPreferenceResponse;
+
+class MercadopagoPreference
+{
+    private $settingsManager;
+
+    public function __construct(
+        SettingsManager $settingsManager)
+    {
+        $this->settingsManager = $settingsManager;
+    }
+
+    /**
+     * @return MercadoPago\Preference
+     */
+    public function __invoke($data, Request $request)
+    {
+        $account = $data->getRestaurant()->getMercadopagoAccount(true);
+        if ($account) {
+            MercadoPago\SDK::setAccessToken($account->getAccessToken());
+        }
+
+        // Create a preference object
+        $preference = new MercadoPago\Preference();
+
+        // Create items for the preference
+        // We add one item for the whole order
+        $items = [];
+        $mpItem = new MercadoPago\Item();
+        $mpItem->title = sprintf('Pedido #%s', $data->getNumber()); // este texto es el que el vendedor puede ver en el detalle de la Operación/Actividad en mercadopago.com.ar/activities/detail/
+        $mpItem->description = $data->getRestaurant()->getName(); // este texto es el que el usuario/comprador puede ver al momento de pagar en el flujo de MP
```

Comment:
PHPStan is complaining because properties are protected, however the `MercadoPago\Entity` class indeed has a magic `__set` method. I think this needs to be ignored in `phpstan.neon` file.

Reply:
I think this should be fixed [here](https://github.com/coopcycle/coopcycle-web/pull/2826/commits/f3e8bf5fb52d8b5474ac2f55d028dedf3e06e1ff?authenticity_token=VVTnGagoDcjeoGxyNtTve1tv70wLqkg35F09rw8tiPmOthygOp9AdW5u%2BpafOO9Q0YbZyKLxHS6Kh84xjqgYyA%3D%3D&file-filters%5B%5D=.neon&file-filters%5B%5D=.php#diff-0361f0c81f363476ddc6f44ab36fcbe66ee685d5f4c2a46b054924591544b766)